src/FingerWhoIs.Lib/FIngerEngine.cs

```diff
@@ -61,6 +61,13 @@ public int CompareTwoFmds(ReadOnlySpan<byte> fmd1, ReadOnlySpan<byte> fmd2,FmdFo
         public List<int> IdentifyFmd(ReadOnlySpan<byte> fmd1,ReadOnlySpan<byte[]> fmds,uint maxToReturn = 10,uint threshold = 21474, FmdFormat format = FmdFormat.Iso)
         {
             var fmdsArray = fmds.ToArray();
+            for (int i = 0; i < fmdsArray.Length; i++)
+            {
+                if (fmdsArray[i] == null)
+                {
+                    fmdsArray[i] = new byte[0];
+                }

```

Comment:
Can you use `is` keyword instead ? to check if stuff is null ?, `==` the equality operator might not always give you a desired result.